chingu-voyages/v51-tier3-team-33
# Feature/refactor navbar

# Description
Added responsiveness to the sidebar. In mobile view the sidebar is hidden and can be accessed by clicking on a hamburger menu in the header. Also added a search bar, logout button and logout confirmation dialog components. 

## Related to the following Jira issue 

https://radhika-gandi.atlassian.net/browse/CT3PB-19?atlOrigin=eyJpIjoiMDZjYmNkZjExMGEwNDE2Njk2NmZiYzEzZDkwZDk1MWQiLCJwIjoiaiJ9

## Main changes explained:
- Added components to app/(dashboard)/ui for mobile view of the sidebar
- Added dialog component for confirmation window

## How to test:
1. check into current branch
2. do  `pnpm install` and `pnpm run dev` to run this PR locally
3. go to /home
4. turn on mobile view
5. verify that the sidebar can be accessed in mobile view via the hamburger icon on top
6. verify that you see the design below
7. click logout button and verify that the dialog window appears
8. click 'cancel', verify that this closes the dialog
9. HELP WANTED on the logout functionality

## Screenshots or videos of changes:


![Screen Shot 2024-09-19 at 11 57 13](https://github.com/user-attachments/assets/dc0681e9-23ea-4984-8c4b-d347d152bab1)

![Screen Shot 2024-09-19 at 12 14 27](https://github.com/user-attachments/assets/b4743f78-cd33-42a0-8b8c-ba0b87eee083)

![Screenshot 2024-09-20 at 3 57 41 PM](https://github.com/user-attachments/assets/99340a56-e4b2-4020-999a-3a680f7dfb00)


## Note:
Any comments will be appreciated! Thanks for reviewing my PR!
'Log out' button does not work yet.
